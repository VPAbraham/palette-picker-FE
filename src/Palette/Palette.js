import React, { Component } from 'react';
import Color from '../Color/Color';
import './Palette.scss';

class Palette extends Component {
  constructor() {
    super()
    this.state = {
      color1: '',
      color2: '',
      color3: '',
      color4: '',
      color5: ''
    }
  }

  componentDidMount() {
    const { color1, color2, color3, color4, color5 } = this.props;
    this.setState({
      color1,
      color2,
      color3,
      color4,
      color5
    })
  }

  static getDerivedStateFromProps(nextProps, prevState){
    const { color1, color2, color3, color4, color5 } = nextProps;
    if(nextProps.color1 !== prevState.color1) {
      return {
        color1: color1,
        color2: color2,
        color3: color3,
        color4: color4,
        color5: color5,
      }
    }
  }
  

  render() {
    return (
      <section className="palette">
        <Color position="color1" color={this.state.color1}/>
        <Color position="color2" color={this.state.color2}/>
        <Color position="color3" color={this.state.color3}/>
        <Color position="color4" color={this.state.color4}/>
        <Color position="color5" color={this.state.color5}/>
      </section>
    )
  }
}

export default Palette;
