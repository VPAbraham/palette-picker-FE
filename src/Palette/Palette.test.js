import React from 'react';
import { Palette } from './Palette';
import { shallow } from 'enzyme';

describe('Palette', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Palette
      color1={'#baddc3'}
      color2={'#d8ac97'}
      color3={'#993f42'}
      color4={'#b4r35'}
      color5={'#ff6767'}
      key='palette'      
    />)
  });
  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});