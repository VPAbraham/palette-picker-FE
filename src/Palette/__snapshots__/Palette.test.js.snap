// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette should match the snapshot with all data passed in 1`] = `
<section
  className="palette"
>
  <Color
    color="#baddc3"
    key="#baddc3"
    position="color1"
  />
  <Color
    color="#d8ac97"
    key="#d8ac97"
    position="color2"
  />
  <Color
    color="#993f42"
    key="#993f42"
    position="color3"
  />
  <Color
    color="#b4r35"
    key="#b4r35"
    position="color4"
  />
  <Color
    color="#ff6767"
    key="#ff6767"
    position="color5"
  />
</section>
`;
